File: consumer/src/main/java/ru/romanow/protocols/consumer/service/PingServiceImpl.java
package ru.romanow.protocols.consumer.service;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

@Service
@RequiredArgsConstructor
public class PingServiceImpl
        implements PingService {
    private static final String PING_PATH = "/api/ping";

    private final RestTemplate restTemplate;

    @Value("${consumer.address}")
    private String consumerAddress;

    @Override
    public boolean ping() {
        ResponseEntity<Void> response =
                restTemplate.getForEntity(consumerAddress + PING_PATH, Void.class);

        return response.getStatusCode().is2xxSuccessful();
    }
}

File: consumer/src/main/java/ru/romanow/protocols/consumer/service/PingServiceImpl.kt
package ru.romanow.protocols.consumer.service

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.springframework.web.client.RestClientResponseException
import org.springframework.web.client.RestTemplate

@Service
class PingServiceImpl(
    @Value("\${consumer.address}")
    private val consumerAddress: String,
    private val restTemplate: RestTemplate,
) : PingService {
    private val logger = LoggerFactory.getLogger(PingService::class.java)

    override fun ping(): Boolean {
        return try {
            return restTemplate.getForObject(consumerAddress + PING_PATH, Boolean::class.java)!!
        } catch (exception: RestClientResponseException) {
            logger.warn(
                "Request to '{}' finish with error {}:{}", PING_PATH,
                exception.rawStatusCode, exception.statusText
            )
            false
        }
    }

    companion object {
        private const val PING_PATH = "/api/v1/ping"
    }
}